### ISSUE TITLE
pick_by has different behavior with iteratee=None compared to lodash

### ISSUE THREAD

**Issue by tailsu** [dgilland/pydash#129]
In Lodash the default iteratee is `_.identity` (for all fooBy functions) so `_.pickBy(x)` returns a dictionary with only those keys that have truthy values. I can't figure out what `pydash.pick_by(x)` does. It seems to always return an empty dictionary.

**Comment by dgilland** [comment]
That is indeed what it does which doesn't seem very useful. So `pydash.pick_by(obj)` should be modified to return a copy of the object instead of an empty dictionary since it's pretty worthless otherwise.

**Comment by tailsu** [comment]
I'd prefer if pick_by (and all foo_by) functions retain the lodash's semantics when called without an explicit iteratee, namely use _.identity.

**Comment by dgilland** [comment]
Most of the other *_by() functions have the expected behavior for iteratee=None but for whatever reason pick_by was missed which should definitely be fixed.

**Comment by elijose55** [comment]
Can I work on this issue?

**Comment by dgilland** [comment]
@elijose55 Go for it!

**Comment by elijose55** [comment]
Should I change the behavior for iteratee=None of the pick_by function to return a copy of the object?

**Comment by dgilland** [comment]
I think the implementation just needs to have something like the following in it:

```python
if iteratee is None:
    iteratee = pyd.identity
    argcount = 1
```

So that when no iteratee is given, it will use the `pyd.identity` as the iteratee.

**Comment by elijose55** [comment]
I tried this implementation and the function returns a copy of the object when the iteratee=None. Thats what is expected right? And one test failed with this:
```sh
tests/test_objects.py::test_pick_by[case2-expected2] FAILED
```

```
case = ([1, 2, 3],), expected = {}

    @parametrize('case,expected', [
        (({'a': 1, 'b': 2, 'c': 3}, ['a', 'b']), {'a': 1, 'b': 2}),
        (({'a': 1, 'b': 2, 'c': 3}, lambda value, key: key in ['a']),
         {'a': 1}),
        (([1, 2, 3],), {}),
        (([1, 2, 3], [0]), {0: 1}),
        ((fixtures.Object(a=1, b=2, c=3), 'a'), {'a': 1}),
        ((fixtures.ItemsObject({'a': 1, 'b': 2, 'c': 3}), 'a'), {'a': 1}),
        ((fixtures.IteritemsObject({'a': 1, 'b': 2, 'c': 3}), 'a'), {'a': 1}),
    ])
    def test_pick_by(case, expected):
>       assert _.pick_by(*case) == expected
E       assert {0: 1, 1: 2, 2: 3} == {}
E         Left contains 3 more items:
E         {0: 1, 1: 2, 2: 3}
E         Full diff:
E         - {0: 1, 1: 2, 2: 3}
E         + {}

tests/test_objects.py:495: AssertionError
```
This is supposed to happen as the pick_by output changed, correct? So now this test should expect the function to just return the object.

**Comment by dgilland** [comment]
Right, that would be the new expected behavior.

**Comment by elijose55** [comment]
Okay. Should I change the test?

**Comment by dgilland** [comment]
Yes, that test would need to be updated.

**Comment by slallum** [comment]
This issue should be closed because PR #139 was merged, right?
